components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -889,6 +898,158 @@ cast_as_duration!(
 );
 cast_as_duration!(Json, cast_json_as_duration, val.unquote()?.as_bytes());
 
+#[rpn_fn(capture = [ctx, extra])]
+fn cast_int_as_time(
+    ctx: &mut EvalContext,
+    extra: &RpnFnCallExtra<'_>,
+    val: &Option<Int>,
+) -> Result<Option<Time>> {
+    if let Some(val) = *val {
+        // Parse `val` as a `u64`
+        Time::parse_from_u64(
+            ctx,
+            val as u64,
```

Comment:
What behavior does MySQL have if the value is a negative number?

Reply:
return `NULL`

Reply:
I think we should check the `val` range (There is an error if the `val` less than 101, which includes negative number). See: https://github.com/pingcap/tidb/blob/master/types/time.go#L1278:L1278

Reply:
I think we should check the `val` range (There is an error if the `val` less than 101, which includes negative number). See: https://github.com/pingcap/tidb/blob/master/types/time.go#L1278:L1278